### DISCUSSION TITLE
Template literal into options like `--user`

### DISCUSSION THREAD

**Discussion by fml09** [Orange-OpenSource/hurl#2577] [Q&A] [answered]
Hello. I would like to insert template strings into options like Basic as described below. It seems that the current latest version of hurl does not support this feature. Is there a chance that this functionality might be added in the future?

```hurl
GET https://{{URL}}/
HTTP 200

[BasicAuth]
{{key}}: {{secret}}
```

**Comment by jcamiel** [comment, marked as answer]
Hi @fml09 ,

Template for basic auth is supported. The syntax is:

```hurl
POST http://foo.com/login
[BasicAuth]
{{user}}: {{password}}
HTTP 200
```

The `[BasicAuth]` is part of the request specification and must be before `HTTP 200` which is the marker of response section.

**Reply by fml09** [reply]
Thank you @jcamiel , it works well! By the way, I have one more question. Is there a way to insert `USERNAME:PASSWORD` as a template into the User option? How should I handle cases like testing with AWS?

```md
POST https://sts.amazonaws.com/
x-amz-security-token: {{TOKEN}}


[Options]
aws-sigv4: {{TEST}}
user: {{USER}}:{{PASSWORD}}

[FormParams]
Action: GetCallerIdentity
Version: 2011-06-15


```

Result:
```
error: Parsing option
  --> test.hurl:7:1
   |
 7 | user: {{USER}}:{{PASSWORD}}
   | ^ the option name is not valid. Valid values are aws-sigv4, cacert, cert, compressed, connect-to, delay, insecure, http1.0, http1.1, http2, http3, ipv4, ipv6, key, location, max-redirs, output, path-as-is, proxy, resolve, retry, retry-interval, skip, unix-socket, variable, verbose, very-verbose
```

**Reply by jcamiel** [reply]
Yes, `--user` is not yet implemented in options sections. For your use case, you can combine with a `[BasicAuth]` section, it _should_ work. Tell me otherwise!

**Reply by fml09** [reply]
```
POST https://sts.amazonaws.com/
x-amz-security-token: {{TOKEN}}

[BasicAuth]
{{user}}: {{password}}

[Options]
aws-sigv4: {{TEST}}

[FormParams]
Action: GetCallerIdentity
Version: 2011-06-15
```

I tested it and it doesn't seem to work

**Reply by jcamiel** [reply]
Thanks I'll look into it and see if we log a bug to fix it. Meanwhile, you can use `--user` on command line as a workaround.

**Comment by jcamiel** [comment]
Hi @fml09 

With #2585, you can now use `--user` in `[Options]` section and you should be able to use template:

```hurl
POST https://sts.amazonaws.com/
x-amz-security-token: {{TOKEN}}
[Options]
aws-sigv4: {{TEST}}
user: {{user}}:{{password}}
[FormParams]
Action: GetCallerIdentity
Version: 2011-06-15
HTTP 200
```

**Reply by jcamiel** [reply]
This will be available in the next Hurl release (4.3.0)

**Reply by fml09** [reply]
Thank you! It works well :)
